internal/controller/eventing/integrationtests/without_apirule_crd/integration_test.go

```diff
@@ -0,0 +1,103 @@
+package controller_switching
+
+import (
+	"os"
+	"testing"
+
+	eventingv1alpha1 "github.com/kyma-project/eventing-manager/api/v1alpha1"
+	"github.com/kyma-project/eventing-manager/test/matchers"
+	"github.com/kyma-project/eventing-manager/test/utils"
+	testutils "github.com/kyma-project/eventing-manager/test/utils/integration"
+	"github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+)
+
+const (
+	projectRootDir = "../../../../../"
+)
+
+var testEnvironment *testutils.TestEnvironment //nolint:gochecknoglobals // used in tests
+
+// TestMain pre-hook and post-hook to run before and after all tests.
+func TestMain(m *testing.M) {
+	// Note: The setup will provision a single K8s env and
+	// all the tests need to create and use a separate namespace
+
+	// setup env test
+	var err error
+	testEnvironment, err = testutils.NewTestEnvironment(testutils.TestEnvironmentConfig{
+		ProjectRootDir:            projectRootDir,
+		CELValidationEnabled:      true,
+		APIRuleCRDEnabled:         false,
+		ApplicationRuleCRDEnabled: true,
+		NATSCRDEnabled:            true,
+		AllowedEventingCR:         nil,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// run tests
+	code := m.Run()
+
+	// tear down test env
+	if err = testEnvironment.TearDown(); err != nil {
+		panic(err)
+	}
+
+	os.Exit(code)
+}
+
+// Test_EventMesh_APIRule_Dependency_Check tests that when the APIRule CRD is missing in case of EventMesh,
+// the Eventing CR should have Error status with a message. The success case where APIRule exists is
+// covered in integrationtests/controller/integration_test.go.
+func Test_EventMesh_APIRule_Dependency_Check(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name          string
+		givenEventing *eventingv1alpha1.Eventing
+		wantMatches   gomegatypes.GomegaMatcher
+	}{
+		{
+			name: "Eventing CR should error state due to APIRule CRD not available",
+			givenEventing: utils.NewEventingCR(
+				utils.WithEventMeshBackend("test-secret-name1"),
```

Comment:
Shouldn't this util func use the secret name and namespace?

Reply:
the namespace is taken from the Eventing CR.